### ISSUE TITLE
IntellisenseManager adds multiple events to the IntelliSenseEventsHandlerProxy 

### ISSUE THREAD

**Issue by ASayre** [microsoft/poshtools#372]
![image](https://cloud.githubusercontent.com/assets/10080679/7079553/98aecfda-deda-11e4-874d-c98311e59114.png)

```
PowerShellTools.Intellisense.IntelliSenseManager: 16:57:50,215 - 1 - DEBUG - Failed to process completion results.
System.ArgumentOutOfRangeException: Length cannot be less than zero.
Parameter name: length
   at System.String.Substring(Int32 startIndex, Int32 length)
   at PowerShellTools.Intellisense.IntelliSenseManager.<>c__DisplayClass21_0.<IntelliSenseManager_CompletionListUpdated>b__0() in E:\OpenSource\poshtools\PowerShellTools\Intellisense\IntelliSenseManager.cs:line 418
```

This causes many bad things.

**Comment by ASayre** [comment]
[If this doesn't actually cause badness, please holler]

**Comment by ericdevg** [comment]
So far not seeing badness yet, however observing one issue around the first intellisense failed due to bad index on caret position. Might or might not be related, investigating.

**Comment by HuanhuanSunMSFT** [comment]
Fixed in PR #464
